tpm2/test/policy_test.go

```diff
@@ -981,3 +1006,160 @@ func TestPolicyAuthValue(t *testing.T) {
 	}
 
 }
+
+func TestPolicyDuplicatonSelect(t *testing.T) {
+	thetpm, err := simulator.OpenSimulator()
+	if err != nil {
+		t.Fatalf("could not connect to TPM simulator: %v", err)
+	}
+	defer thetpm.Close()
+
+	tests := []struct {
+		name          string
+		IncludeObject bool
+	}{
+		{"IncludeObjectFalse", false},
+		{"IncludeObjectTrue", true},
+	}
+
+	var pav PolicyDuplicationSelect
+	var pgd *PolicyGetDigestResponse
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			ek, ekcleanup := primaryRSAEK(t, thetpm)
+			defer ekcleanup()
+
+			pk, pkcleanup := primaryRSASRK(t, thetpm)
+			defer pkcleanup()
+
+			if tt.IncludeObject {
```

Comment:
This test could be a bit more readable if `if tt.IncludeObject` were used to manipulate only `pav`'s `ObjectName` and `IncludeObject` fields rather than kicking off a bunch of other calls.

Since the gist of this test case is "make sure that multiple flavors of `PolicyDuplicationSelect` all compute the right policy hash", we should pretty much do the same setup in all cases, and just do a minor tweak on `pav` before execution depending on the test case.

Reply:
ok, i think i understand; ptal the latest commit.